cmd/query/app/server_test.go

```diff
@@ -881,32 +881,30 @@ func TestServerHTTP_TracesRequest(t *testing.T) {
 		expectedTrace string
 	}{
 		{
-			name:          "different ports",
+			name:          "different ports (OTEL implementation)",
 			httpEndpoint:  ":8080",
 			grpcEndpoint:  ":8081",
 			queryString:   "/api/traces/123456aBC",
 			expectedTrace: "/api/traces/{traceID}",
 		},
 		{
-			name:          "different ports for v3 api",
+			name:          "different ports (OTEL implementation) for v3 api",
 			httpEndpoint:  ":8080",
 			grpcEndpoint:  ":8081",
 			queryString:   "/api/v3/traces/123456aBC",
 			expectedTrace: "/api/v3/traces/{trace_id}",
 		},
 		{
-			name:          "same port",
-			httpEndpoint:  ":8080",
-			grpcEndpoint:  ":8080",
-			queryString:   "/api/traces/123456aBC",
-			expectedTrace: "/api/traces/{traceID}",
```

Comment:
why are we loosing this trace? That was my point of adding the test to main first - it should not change in this PR.

Reply:
@yurishkuro We're losing this trace at the moment because removed the call to `otelhttp.NewHandler` which was taking a tracer. If we don't remove that call then we run into the issue of double instrumentation that we were seeing earlier.

Reply:
so how do we explain that? The OTEL helper is supposed to add tracing middleware, why is it not working?

Reply:
@yurishkuro The OTEL helper does add tracing. However, when the ports are the same, we're not using the OTEL helper to initialize the server and are our falling back to our custom implementation. We were previously using `otelhttp.NewHandler` which was previously rendering this trace. We removed it from there because then we were running into the case of double instrumentation when the ports were different and we were using the OTEL helpers.

Reply:
Thinking more about this - is there a reason why we can't just OTEL's helper to create the server even if the ports are the same. We can still use a different flow when we start the listener.

Reply:
yes, I was going to say the same - in the Legacy function we should still use otelhttp.NewHandler

Reply:
@yurishkuro I was actually wondering why we can't just use the server created by `ToServer` for the legacy flow as well?

Reply:
How would we co-locate http and grpc servers on the same port in that case? We could if the listener is still done separately by our code.

Reply:
@yurishkuro The listener is initialized separately and it already has a separate check for separate ports (see https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/server.go#L289). So can we just simplify the logic of creating the server to always create it using `ToServer` and then we can handle the legacy behaviour inside `ToListener`.

Reply:
give it a try. Or just add tracing handler to Legacy function and we can merge this PR, then try the listener separately.

Reply:
@yurishkuro I removed the bifurcation for the creation of the server and it looks to be working as expected which makes sense. I'll do the same for the GRPC server in a follow-up PR.